Documentation/MixedRealityConfigurationGuide.md

```diff
@@ -359,23 +353,99 @@ using UnityEngine;
+
+A noticable application hesitation may occur during this process. Also any script with higher priority than the `MixedRealityToolkit` script can enter its Update before the new profile is properly setup. See [Script Execution Order settings](https://docs.unity3d.com/Manual/class-MonoManager.html) for more information on script priority.
+
+Special handling is required when Unity UI elements are present in the scene. Such elements are usually children of `Canvas` that often requires a reference to the world camera (UI camera). During an active profile switch, UI cameras are destroyed and recreated and thus the references to them have to be updated for all canvases with previously assigned world camera. `VerifyCanvasConfiguration` in the `CanvasUtility` script can be used to update these references. The following script is an example of one way to handle this problem.
+
+
+```csharp
+/// <summary>
+/// Sample MonoBehaviour that will change the active profile and update the references
+/// to UI camera for Canvases to ensure the functionality of Unity UI after the change.
+/// </summary>
+/// <remarks>
+/// Note that this script assumes there is no instantiation and/or destroy of Canvases.
+/// Canvases to be used with MRTK have the CanvasUtility script attached.
+/// </remarks>
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ActiveProfileSwapper : MonoBehaviour
+{
+    
+    public MixedRealityToolkitConfigurationProfile ProfileToUse;
+    private CanvasUtility[] allCanvases;// All MRTK canvases (i.e. with the CanvasUtility script) in scene
+    private List<CanvasUtility> cameraNotNullCanvases; // Canvases with not-null cameras
+
+    public void ChangeActiveProfile()
+    {
+        if (cameraNotNullCanvases == null)
+        {
+            // Find canvases with not-null world cameras
+            cameraNotNullCanvases = new List<CanvasUtility>();
+            foreach (var canvas in allCanvases)
+            {
+                if (canvas.GetComponent<Canvas>().worldCamera != null)
+                {
+                    cameraNotNullCanvases.Add(canvas);
+                }
+            }
+        }
+        MixedRealityToolkit.Instance.ActiveProfile = ProfileToUse;
+        StartCoroutine(UpdateCanvasCameraReference());
+    }
+
+    private IEnumerator UpdateCanvasCameraReference()
+    {
+        // Wait for the profile change to complete
+        yield return null; // Skip the frame the profile swap started
+        yield return null; // Skip the next frame as well as the UI camera creation may still be in process
```

Comment:
Similar to how you've updated the input module to persist between profiles, I wonder if we want the UI raycast camera to also persist? Since it's a Unity component, I'm not sure it has any dependencies itself on the active MRTK instance. The FocusProvider already has code to find an existing raycast camera on startup:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/e07020b0e84340be604956a543cb54ce88e28d4f/Assets/MRTK/Services/InputSystem/FocusProvider.cs#L693-L701

Would we want to suppress its clean-up during a profile switch, like you did with the input module?

Reply:
The cleanup time behavior of UI camera is currently different than the input module mainly because we expect the module to be in the scene when we start playing the scene but UI camera is always created by MRTK at start. As it is created by MRTK it may be logical to make it go away during the destroy process. We could take the other route, but it will take much time to fully retest the PR (all combinations of pipeline/hardware/remote vs on device etc.)

Reply:
That works. I think it'd be worth filing an issue to see if we can handle that camera better during a reload, but this code snippet and doc should be sufficient until then.